### PR TITLE
New Tool Qualities/Tools to use them

### PR DESCRIPTION
**Pull request by ddsaw**
#### Summary
```SUMMARY: Features "Adds new tool qualities as well as tools that use them/modifying some existing tools as well as their spawn table"```
#### Purpose of change
Lays some groundwork for #22688 as well as other recipes that may be modified to require these new qualities.

#### Describe the solution
Implementing some of the needed tools. Not all tools described in the homemade expedient firearms book are implemented with this (and neither is the recipe for the luty smg, to begin with) but it's a step forward. Whilst vicing is implemented as a quality, it is not yet used as unlike the others, a vice is not a handheld tool. The same can be said for the bench grinder, though I am not yet sure if that is strictly needed, considering the anglegrinder exists.
The setting square is I believe too minute to require it's own quality so I'm not yet sure what to do with it. The "taper cutting tool" is something I'm not entirely sure of, as in, what it is.
I believe the countersink is something that should be abstracted into already existing drills, unless we want to simulate a plethora of different drillheads going forward.
Currently, none of these tools have craftable versions and they must all be found. They are not used in any recipes yet, either.

#### Describe alternatives you've considered
I've considered implementing vicing as a "requires nearby vice" thing going forward instead of the quality it is now. I've considered giving all drills level 1 reaming in general. I've considered adding reaming to swiss army knives, because some models do have a reamer.

### CODE REVIEW

File: data/json/items/tools.json [hunk 1]
```diff
@@ -9166,5 +9170,64 @@
     "name": "metal hand axe",
     "description": "This is a chunk of steel with one edge hammered down to something resembling a cutting edge.  It works passably well as an axe but really can't compare to a proper axe.",
     "material": "steel"
+  },
+  {
+    "id": "pin_reamer",
+    "type": "TOOL",
+    "name": "pin reamer",
+    "description": "Handheld pin reamers of this kind are used to enlargen existing holes, or remove any burs and such from them",
```
Comment (Night-Pryanik) [inline data/json/items/tools.json]:
```suggestion
    "description": "Handheld pin reamers of this kind are used to enlarge existing holes, or remove any burs and such from them.",
```

File: data/json/items/tools.json [hunk 2]
```diff
@@ -9166,5 +9170,64 @@
...
+    "material": "steel",
+    "symbol": ";",
+    "color": "light_gray",
+    "qualities": [ [ "REAM", 2 ] ]
+  },
+  {
+    "id": "metal_file",
+    "type": "TOOL",
+    "name": "metal fileset",
+    "description": "These tools are commonly used to remove small amounts of materials from the surface of metal objects. ",
```
Comment (Night-Pryanik) [inline data/json/items/tools.json]:
```suggestion
    "description": "These tools are commonly used to remove small amounts of materials from the surface of metal objects.",
```

File: data/json/tool_qualities.json [hunk 3]
```diff
@@ -211,5 +211,25 @@
     "name": "chromatography",
     "//": "Think of this as 'fine separating'. Chromatography represents a number of precise separation techniques",
     "//": "useful in both analysis and purification."
+  },
+  {
+    "type": "tool_quality",
+    "id": "VICE",
```
Comment (Night-Pryanik) [inline data/json/tool_qualities.json]: This tool quality isn't used anywhere.

Comment (ddsaw) [reply data/json/tool_qualities.json]: Not yet. I'm currently looking at implementing it. Perhaps even removing it, I'm not so sure myself.

**Pull request merged**